collector/arp_linux.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !noarp
+
+package collector
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type arpCollector struct {
+	count *prometheus.Desc
+}
+
+func init() {
+	Factories["arp"] = NewArpCollector
+}
+
+// NewArpCollector returns a new Collector exposing ARP stats.
+func NewArpCollector() (Collector, error) {
+	return &arpCollector{
+		count: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, "arp",
+				"count"), "ARP entries by device",
+			[]string{"device"}, nil,
+		),
+	}, nil
+}
+
+func getArpEntries() (map[string]uint32, error) {
+	file, err := os.Open(procFilePath("net/arp"))
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	return parseArpEntries(file), nil
+}
+
+func parseArpEntries(data *os.File) map[string]uint32 {
```

Comment:
The only requirement is to pass something implementing the `io.Reader` interface, so I'd reflect that in the function signature.